Essentials/src/com/earth2me/essentials/services/ReserveEssentials.java

```diff
@@ -0,0 +1,966 @@
+ */
+public class ReserveEssentials implements EconomyAPI {
+
+    private final Essentials ess;
+
+    public ReserveEssentials(Essentials plugin) {
+        this.ess = plugin;
+    }
+
+    public static void register(Essentials plugin) {
+        //Check to see if there is an economy provider already registered, if so Essentials will take the back seat.
+        if (!((Reserve) Bukkit.getServer().getPluginManager().getPlugin("Reserve")).economyProvided()) {
+            Reserve.instance().registerProvider(new ReserveEssentials(plugin));
+        }
+    }
+
+    /**
+     * @return The name of the Economy implementation.
+     */
+    @Override
+    public String name() {
+        return "EssentialsX";
+    }
+
+    /**
+     * @return The version of Reserve the Economy implementation supports.
+     */
+    @Override
+    public String version() {
+        return "0.1.4.6";
+    }
+
+    //This is our method to convert UUID -> username for use with Essentials' create account methods.
+    private String getName(UUID identifier) {
+        final User user = ess.getUser(identifier);
+        return ((user == null) ? identifier.toString() : user.getName());
+    }
+
+    /**
+     * @return Whether or not this implementation is enabled.
+     */
+    @Override
+    public boolean enabled() {
+        return true;
+    }
+
+    /**
+     * @return Whether or not this implementation should have a default Vault implementation.
+     */
+    @Override
+    public boolean vault() {
+        return false;
+    }
+
+    /**
+     * Used to get the plural name of the default currency.
+     * @return The plural name of the default currency.
+     */
+    @Override
+    public String currencyDefaultPlural() {
+        return tl("moneyPlural");
+    }
+
+    /**
+     * Used to get the singular name of the default currency.
+     * @return The plural name of the default currency.
+     */
+    @Override
+    public String currencyDefaultSingular() {
+        return tl("moneySingular");
+    }
+
+    /**
+     * Used to get the plural name of the default currency for a world.
+     * @param world The world to be used in this check.
+     * @return The plural name of the default currency.
+     */
+    @Override
+    public String currencyDefaultPlural(String world) {
+        return tl("moneyPlural");
+    }
+
+    /**
+     * Used to get the singular name of the default currency for a world.
+     * @param world The world to be used in this check.
+     * @return The plural name of the default currency.
+     */
+    @Override
+    public String currencyDefaultSingular(String world) {
+        return tl("moneySingular");
+    }
+
+    /**
+     * Checks to see if a {@link Currency} exists with this name.
+     * @param name The name of the {@link Currency} to search for.
+     * @return True if the currency exists, else false.
+     */
+    @Override
+    public boolean hasCurrency(String name) {
+        return name.equalsIgnoreCase(tl("moneySingular"));
```

Comment:
are currencies case insensitive?

Reply:
Yes. Dollar is the same as dollar in the real world.